nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d.hpp

```diff
@@ -407,27 +408,39 @@ class Costmap2D
     int offset_dx = sign(dx);
     int offset_dy = sign(dy) * size_x_;
 
-    unsigned int offset = y0 * size_x_ + x0;
-
     // we need to chose how much to scale our dominant dimension,
     // based on the maximum length of the line
     double dist = std::hypot(dx, dy);
-    double scale = (dist == 0.0) ? 1.0 : std::min(1.0, max_length / dist);
+    if (dist < min_length) {
+      return;
+    }
 
+    // Adjust x0, y0 and offset to start from min_length distance
+    x0 = (unsigned int)(x0 + dx / dist * min_length);
```

Comment:
Don't override variables with new quantities. Make a new variable and name it so this can be understood